From 5f500047be0fd90f23e959d1a7775e7fa6b23999 Mon Sep 17 00:00:00 2001
From: Nuno Silva <little.coding.fox@gmail.com>
Date: Mon, 4 Oct 2021 18:04:23 +0100
Subject: [PATCH] Add dynamic linking

---
 gcc/config/toast.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gcc/config/toast.h b/gcc/config/toast.h
index 0b1e69a0448..d88485c8c9e 100644
--- a/gcc/config/toast.h
+++ b/gcc/config/toast.h
@@ -3,7 +3,7 @@
 #define TARGET_TOAST 1
  
 /* Default arguments you want when running your
-   i686-myos-gcc/x86_64-myos-gcc toolchain */
+   i686-myos-gcc/x86_64-toast-gcc toolchain */
 #undef LIB_SPEC
 #define LIB_SPEC "-lc" /* link against C standard library */
  
@@ -11,6 +11,9 @@
    The %s tells GCC to look for these files in the library directory. */
 #undef STARTFILE_SPEC
 #define STARTFILE_SPEC "crt0.o%s crti.o%s crtbegin.o%s"
+
+#undef LINK_SPEC
+#define LINK_SPEC "%{shared:-shared} %{static:-static} %{!shared: %{!static: %{rdynamic:-export-dynamic}}}"
  
 /* Files that are linked after user code. */
 #undef ENDFILE_SPEC
@@ -22,7 +25,7 @@
   do {                                \
     builtin_define ("__toast__");      \
     builtin_define ("__unix__");      \
-    builtin_assert ("system=myos");   \
+    builtin_assert ("system=toast");   \
     builtin_assert ("system=unix");   \
     builtin_assert ("system=posix");   \
   } while(0);
\ No newline at end of file
-- 
2.30.2.windows.1

